### ISSUE TITLE
Clock gating capability in BOOM

### ISSUE THREAD

**Issue by owahidcse99** [riscv-boom/riscv-boom#472]
**Type of issue**: bug report | feature request | question | other enhancement
Feature request

**Impact**: no rtl change | rtl refactoring | new rtl | unknown
new rtl

**Development Phase**: request | proposal
proposal
**Other information**

**What is the current behavior?**
No clock gating
**What is the expected behavior?**
Clock gating
**Please tell us about your environment:**
Rocket core has some clock gating capability in Front-end
https://github.com/chipsalliance/rocket-chip/blob/1872f5d501221f13950aa2293939634a1e0d1735/src/main/scala/rocket/Frontend.scala

icache.clock := gated_clock
  icache.io.clock_enabled := clock_en
  withClock (gated_clock) { // entering gated-clock domain

I wonder if BOOM could implement some clock gating features so different pipeline stages could run at different clock domains based on some conditions. I am not much familiar to BOOMv3 but I simulated BOOMv2 and it does not support clock gating. If BOOM targets low power segment, then gating clock is a must for low activity region without penalizing IPC. Any consideration is much appreciated.
Thanks

**What is the use case for changing the behavior?**

**Comment by jerryz123** [comment]
I agree clock-gating is necessary for power efficient applications. Following the Rocket implementation of ICache clock gating should be a good starting point for implementing clock-gating in BOOM's ICache.
